lint/linter/test-spec-urls.ts

```diff
@@ -49,19 +54,62 @@ const specsExceptions = [
   'https://github.com/WebAssembly/js-promise-integration',
 ];
 
-const allowedSpecURLs = [
-  ...(specData
-    .filter((spec) => spec.standing == 'good')
-    .map((spec) => [
-      spec.url,
-      spec.nightly?.url,
-      ...(spec.nightly ? spec.nightly.alternateUrls : []),
-      spec.series.nightlyUrl,
-    ])
-    .flat()
-    .filter((url) => !!url) as string[]),
-  ...specsExceptions,
-];
+interface ValidSpecHosts {
+  url: string;
+  alternateUrl: string;
+}
+
+const validSpecHosts: ValidSpecHosts[] = [];
+
+/**
+ * Get valid specification URLs from webref ids
+ * @returns array of valid spec urls (including fragment id)
+ */
+const getValidSpecURLs = async (): Promise<string[]> => {
+  const indexFile = await fetch(
+    'https://raw.githubusercontent.com/w3c/webref/main/ed/index.json',
+  );
+  const index = JSON.parse(await indexFile.text());
+  const specIDs: string[] = [];
+
+  index.results.forEach((spec) => {
+    if (spec.standing === 'good') {
+      if (spec.shortname === 'webnn') {
+        validSpecHosts.push({
+          url: spec.series.releaseUrl,
+          alternateUrl: spec.nightly?.url,
+        });
+      } else {
+        validSpecHosts.push({
+          url: spec.series.nightlyUrl,
+          alternateUrl: spec.nightly?.url,
+        });
+      }
+      if (spec.ids) {
+        specIDs.push(spec.ids);
+      }
+    }
+  });
+
+  const specURLsWithFragments: string[] = [];
+  await Promise.all(
+    specIDs.map(async (id) => {
+      const idFile = await fetch(
+        `https://raw.githubusercontent.com/w3c/webref/main/ed/${id}`,
+      );
+      const idResponse = JSON.parse(await idFile.text());
+      specURLsWithFragments.push(
+        ...idResponse.ids.map((id) => {
+          const url = new URL(id);
+          return url.origin + url.pathname + decodeURIComponent(url.hash);
+        }),
+      );
+    }),
+  );
+  return specURLsWithFragments;
+};
```

Comment:
Before we merge this, we need to find a solution that avoids these downloads.

I'm not sure if it really makes sense to publish `@w3c/ed` as an NPM package, because that's a lot of data, and we only need the IDs. So we might want to publish something similar to @ddbeck's [mdn-content-inventory](https://github.com/ddbeck/mdn-content-inventory) instead.

Reply:
Yeah, having something similar to Daniel's inventory might be great! Or we combine them so that is only one daily package for updates that does both? Not sure.

In the meantime, should we land the fixes you identified to spec_urls in a separate PR?